### PR TITLE
Add config file support, refactorings

### PR DESCRIPTION
**Pull request by deadtrickster**
Hi,

also features: 
- optional verbose indent error/stacktrace;
- indent options depend on indentW.

Today I spent some time integrating erlang_ls -i with my emacs: https://github.com/deadtrickster/erlang-format. 

How to ask indent to remove traling whitespace?

### CODE REVIEW

**Comment by deadtrickster** [comment]
Oh, there is no tests. for the new features. Please let me know what's the best way to test config file.

File: apps/erlang_ls/src/erlang_ls.erl [hunk 1]
```diff
@@ -26,7 +29,8 @@ cli_options() ->
      {dump,    $d,        "dump",     string,      "Dump sourcer db for file"},
      {port,    $p,        "port",    integer,               "LSP server port"},
      {verbose, $v,        "verbose", integer,               "Verbosity level"},
-     {indent,  $i,        "indent",   string,      "Indent file(s) and exit"}
+     {indent,  $i,        "indent",   string,      "Indent file(s) and exit"},
+     {config,  undefined,  "config",   string,      "Indent file(s) and exit"}
```
Comment (vladdu) [inline apps/erlang_ls/src/erlang_ls.erl]: Please correct the 'config' option's description.

**Comment by vladdu** [comment]
This looks good, thank you Ilya!  There are a few minor points:
- description for the 'config' option doesn't match
- I am not sure if the comments for indentation preferences (and thus the use of IndentW) are entirely correct, I think nobody tested with other values than 4. I'm not sure if other values make much sense. Did you try any other value to see if the result looks alright?
- please document the config file's format. I think the right place would be in the README file, when there will be a description of the CLI usage there. For the time being, I think just add it there as a side note.

I don't think there is support for removing the trailing whitespace. Usually, editors have that kind of functionality in a generic way, but I suppose it wouldn't hurt to do it ourselves anyway. Please open a ticket for that.

**Comment by deadtrickster** [comment]
> Did you try any other value to see if the result looks alright?

yep. I use indentW=2 in my own projects, the only thing I noticed (so far) is difference for 'before_arrow'. 

But, since config added that can override all paramters I would live with what we have and expect people to patch this naturally. I assume for indentW=4 it behaves as erlang-mode anyway.

**Comment by vladdu** [comment]
It may be that 'before_arrow' should be 2, instead of IndentW div 2?

**Comment by deadtrickster** [comment]
dunno, now I think it makes sense as it visually distinguishes between function body and when

**Comment by vladdu** [comment]
Ok, no problem. Looks good now, if there are problems later we can change those. It's great that you did the emacs integration, we can get feedback from there too.

**Comment by vladdu** [comment]
Oh, yes, testing the config file... I guess only getPrefs should be tested, because maybe_load_config is just a wrapper around file:consult. We are interested in checking that changing IndentW results in changed indentation options (only checking a couple is enough).

**Comment by deadtrickster** [comment]
added a test

**Comment by vladdu** [comment]
Thank you very much!

**Pull request merged**